src/main/java/org/folio/rest/impl/OaiPmhImpl.java

```diff
@@ -87,7 +92,19 @@ public void getOaiIdentifiers(String resumptionToken, String from, String until,
   @Override
   public void getOaiMetadataFormats(String identifier, Map<String, String> okapiHeaders,
                                     Handler<AsyncResult<Response>> asyncResultHandler, Context vertxContext) {
-    asyncResultHandler.handle(succeededFuture(GetOaiMetadataFormatsResponse.respond500WithTextPlain("The verb ListMetadataFormats is not supported yet :(")));
+
+    Request request = Request.builder().identifier(identifier).build();
```

Comment:
Request holds okapi headers among other things. Since httpClient is only needed in helper, I think it's better to create it inside helper.

Reply:
RestVerticle uses [CaseInsensitiveMap](https://commons.apache.org/proper/commons-collections/apidocs/org/apache/commons/collections4/map/CaseInsensitiveMap.html) to collect headers so once data added there all the keays are converted to lower case. However the Request builder creates copy of the headers `ImmutableMap.copyOf(okapiHeaders)`. After this the values can be only retrieved if the key is converted to lower case manually

Reply:
I agree, the Request should not replace map implementation. I think it is already fixed in Piotr's PR.